rasa/shared/nlu/training_data/training_data.py

```diff
@@ -679,38 +678,54 @@ def can_train_nlu_model(self) -> bool:
         ]
         return not any([len(lst) > 0 for lst in lists_to_check])
 
-    def divide_into_chunks(self, max_size: int = 2) -> List["TrainingDataChunk"]:
+    def divide_into_chunks(self, num_chunks: int) -> List["TrainingDataChunk"]:
         """Divides the training data into smaller chunks.
 
         Each chunk should be a good representation of the complete dataset. E.g. it
         should not just contain examples of one intent, but instead match the
         distribution of the complete dataset.
 
         Args:
-            max_size: The maximum size (in MB) of one chunk.
+            num_chunks: The total number of chunks into which the training data should be broken.
 
         Returns:
             A list of all training data chunks.
         """
-        pass
+        all_chunks = []
+        data_to_chunk = copy.deepcopy(self)
 
+        for chunk_index in range(num_chunks):
 
-class TrainingDataChunk:
-    """Holds a portion of the complete TrainingData."""
+            chunk_size_fraction = 1 / (num_chunks - chunk_index)
+            current_chunk, leftover_examples = data_to_chunk.split_nlu_examples(
+                1 - chunk_size_fraction
+            )
 
-    def __init__(
-        self,
-        training_examples: List[Message],
-        responses: Optional[Dict[Text, List[Dict[Text, Any]]]] = None,
-    ) -> None:
-        """Initialize a training data chunk.
+            if chunk_index == num_chunks - 1:
```

Comment:
I guess we could move this check up and avoid calling `split_nlu_examples` on the last chunk. We could just take all remaining examples directly.

Reply:
Removed 👍